### DISCUSSION TITLE
RTSP Stream Issue

### DISCUSSION THREAD

**Discussion by clee604** [gtxaspec/wz_mini_hacks#121] [Q&A]
Does anyone else see some artifacts or FPS drops with their RTSP streams?

I've noticed the RTSP stream shows some horizonal artifacts occasionally on a HW decoder or FPS drops when on a software decoder (video drops frames and the video skips a second, really noticeable if you have the timestamp on). I noticed this on both wifi and ethernet connected V3 cameras. I have not noticed this on the one V2 camera I have. It is most noticeable in tinycam but I see it in VLC as well (FPS drop and frame skip).

My RTSP config:
`RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_FPS="20"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"`

I've tested with following different parameters to try and resolve it
FPS = 15, 20 or leaving it blank with same results
ENC_PARAMETER= Tried all 2, 4, 8 with same results

**Comment by gtxaspec** [comment]
artifacting is almost exclusively a connectivity issue from my experience and research

changing encoder parameters won't solve it, lowering the bitrate may.  change wifi channels or switch to wired if possible

**Reply by clee604** [reply]
the camera is wired, and I notice the issue doesn't show up in the wyze app

**Reply by gtxaspec** [reply]
I also experienced this on a wired network as well when the bitrate was too high, and the USB Ethernet adapter was poor quality
just fyi

**Reply by clee604** [reply]
I can try to drop the bitrate, but when I raised the bitrate previously it didn't increase the frequency or severity of the artifacts so it led me to believe it's not a bitrate issue

**Reply by clee604** [reply]
I tested the ethernet adapter (to the camera via ssh) with iperf3 its performance is good, so I don't think its a bandwidth issue either
edit: Tested with bitrate cut in half and no difference

**Comment by clee604** [comment]
Did some more testing to help rule out/narrow down the issue:
V3
- Turning rtsp audio off doesn't seem to help, in fact seems to lead to more artifacts
- Lowering bitrate did not improve things
- Wyze app video is fine, only RTSP seems to have issue

V2
- No issues here, even with the bitrate cranked way up

**Comment by gtxaspec** [comment]
set this parameter to: `RTSP_HI_RES_FPS=""`  and see if you still get the artifacts

**Reply by clee604** [reply]
Hi @gtxaspec , I had tried leaving this blank and dropping it to 15 as part of my testing and it still occurs unfortunately
